pkg/smclient/client.go

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package smclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/Peripli/service-manager-cli/pkg/errors"
+	"github.com/Peripli/service-manager-cli/pkg/httputil"
+	"github.com/Peripli/service-manager-cli/pkg/types"
+)
+
+// Client should be implemented by SM clients
+//go:generate counterfeiter . Client
+type Client interface {
+	RegisterPlatform(*types.Platform) (*types.Platform, error)
+	RegisterBroker(*types.Broker) (*types.Broker, error)
+	ListBrokers() (*types.Brokers, error)
+	ListPlatforms() (*types.Platforms, error)
+	DeleteBroker(string) error
+	DeletePlatform(string) error
+	UpdateBroker(string, *types.Broker) (*types.Broker, error)
+	UpdatePlatform(string, *types.Platform) (*types.Platform, error)
+}
+
+type serviceManagerClient struct {
+	config     *ClientConfig
+	httpClient *http.Client
+	headers    *http.Header
+}
+
+// NewClient returns new SM client
+func NewClient(config *ClientConfig) Client {
+	client := &serviceManagerClient{config: config, httpClient: &http.Client{}}
```

Comment:
Why not simply:
client := &serviceManagerClient{config: config, httpClient: &http.Client{}, headers: &http.Header{}}